### DISCUSSION TITLE
Can the Priority of a task be set at time of creation?

### DISCUSSION THREAD

**Discussion by dumontjames** [dahall/TaskScheduler#887] [Q&A] [answered]
Hello,

Is it possible to set the priority of a task when creating it from VB.Net? Currently, I have to do this manually. Once the task is completed, I export the task and change the Priority setting to a lower value. I then delete the task and import using the modified XML file.
Some of the scripts I run take hours to complete and setting the Priority to a lower value speeds the process considerably.

Here's and exampleof the XML file,

    <WakeToRun>false</WakeToRun>
    <ExecutionTimeLimit>PT72H</ExecutionTimeLimit>
    **<Priority>4</Priority>**
  </Settings>

**Comment by dahall** [comment, marked as answer]
```vb
Dim tDefinition As TaskDefinition = TaskService.Instance.NewTask
' This will set the <Priority> to 3, but it does elevate it
tDefinition.Settings.Priority = ProcessPriorityClass.AboveNormal
' This will set the <Priority> to 4, but reads a little less nicely
tDefinition.Settings.Priority = CType(4, ProcessPriorityClass)
```

**Comment by dumontjames** [comment]
Hi David,

Thank you for the quick response. This is exactly what I was looking for.


On Thu, May 20, 2021 at 12:43 PM David Hall ***@***.***>
wrote:

> Dim tDefinition As TaskDefinition = TaskService.Instance.NewTasktDefinition.Settings.Priority = ProcessPriorityClass.AboveNormal
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dahall/TaskScheduler/discussions/887#discussioncomment-764399>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJPNPWMOITO6LTWQICTWDWLTOU32LANCNFSM45GYEG4A>
> .
>


-- 

Thank you,

*James Dumont*

Application Engineer

*MicroCAD Training & Consulting*
*Direct:* 617-321-2415  *Toll Free:* 888-355-0081
*Certified MBE and DBE*
Reality Capture <https://www.microcad3d.com/scanning-services> | Software
<https://www.microcad3d.com/#products> | HP Plotters
<https://www.microcad3d.com/hp-printers/> | BIM Services
<https://www.microcad3d.com/#services> | Training
<https://www.microcad3d.com/class-schedule/>

Follow us on: Facebook
<https://www.facebook.com/microcadtrainingandconsulting> | Youtube
<https://www.youtube.com/MicroCAD3D> | LinkedIn
<https://www.linkedin.com/company/microcad-training-&-consulting> | Twitter
<https://twitter.com/MicroCAD3d>
